src/qt/configurenamedialog.cpp

```diff
@@ -72,6 +74,35 @@ void ConfigureNameDialog::accept()
         if(MessageBoxInvalidJSON == QMessageBox::Ok){
             QDialog::accept();
         }
+
+    } else if (!IsMinimalJSONOrEmptyString(data)) {
+        QMessageBox MessageBoxNonMinimalJSON;
+        MessageBoxNonMinimalJSON.setIcon(QMessageBox::Warning);
+        MessageBoxNonMinimalJSON.setWindowTitle(tr("Non-minimal JSON"));
+        MessageBoxNonMinimalJSON.setText(tr("Are you sure you want to continue anyway? The inputted JSON data is non-minimal, and therefore will waste space as well as incurring added transaction costs when written on the blockchain.")); 
+        MessageBoxNonMinimalJSON.addButton(QMessageBox::Ok);
+        MessageBoxNonMinimalJSON.addButton(QMessageBox::Cancel);
+        MessageBoxNonMinimalJSON.addButton(tr("Minimalise JSON"), QMessageBox::ActionRole);
+        
+        MessageBoxNonMinimalJSON.exec();
+
+        QMessageBox::ButtonRole reply = MessageBoxNonMinimalJSON.buttonRole(MessageBoxNonMinimalJSON.clickedButton());
+
+        if(reply == QMessageBox::AcceptRole){
+            QDialog::accept();
+        } else if(reply == QMessageBox::RejectRole){
+            
```

Comment:
This looks unfinished?  Is this still WIP?

Reply:
It's so it does nothing if a user clicks the Cancel button.

Reply:
I think we can just leave out the `RejectRole` part of the `if`/`else` chain? If the intent is to make it clear that `RejectRole` is a possibility, I think a comment would be sufficient for that.

Reply:
Yes I would suggest to either leave it out, or if you think it needs to be explicitly explained that we want to do nothing in that case, then leave it in but add a comment to that effect.